### ISSUE TITLE
gosdk pipelines with arguments fail to start 

### ISSUE THREAD

**Issue by lonnc** [gaia-pipeline/gaia#292]
Start gaia using docker:
`docker run -d -p 8080:8080 -v $PWD:/data gaiapipeline/gaia:latest`

Create pipeline using http://localhost:8080/#/pipeline/create with fields:
- Repo: https://github.com/gaia-pipeline/go-example
- Branch: refs/heads/parameters_example
- Name: test

Status Output:
```
Git repository has been cloned.
Starting build process. This may take some time...
Pipeline has been successfully compiled.
Verifying pipeline...
Pipeline has been verified.
Finalizing creation process...
Copy was successful.
Finished.
```

1. View pipeline at http://localhost:8080/#/pipeline/detail?pipelineid=1
2. Hit "Start Pipeline" button then populate fields:
    - Username for the database schema: Bob
    - Description for username: Hello World
3. Hit "Start Pipeline" button
4. Fails to start returning the error:
```
Error: 400
invalid content provided in request
```

Pipelines with no arguments are working for me (e.g. https://github.com/gaia-pipeline/go-example Branch refs/heads/master), but any pipelines that I've tried to create which do have an argument are failing to start.

**Comment by Skarlso** [comment]
Hi. Thanks for opening an issue. I'll look into it.

**Comment by Skarlso** [comment]
Oh I see. Yes, the problem is that that's not the only thing needed for the parameter thing to work. You also needs a specific SDK and a different version of Gaia. All in these PRs:

https://github.com/gaia-pipeline/protobuf/pull/2
https://github.com/gaia-pipeline/gosdk/pull/5
https://github.com/gaia-pipeline/gaia/pull/238

And of course this only works for Go based pipelines...

**Comment by Skarlso** [comment]
The parameters branch is most likely outdated by now since I updated a lot of things. I'll do a push into that so you can try again. :)

**Comment by Skarlso** [comment]
@lonnc Please try https://github.com/gaia-pipeline/go-example/tree/parameter_example_v2

**Comment by lonnc** [comment]
Unfortunately still getting the same error when trying to start the pipeline:
- Repo: https://github.com/gaia-pipeline/go-example
- Branch: refs/heads/parameter_example_v2

```
Error: 400
invalid content provided in request
```

Available logs are below:
```
2020-11-27T17:15:45.828Z [INFO ] Gaia: worker gRPC server about to start on port: 8989
2020-11-27T17:16:32.999Z [DEBUG] plugin: starting plugin: path=/data/tmp/golang/src/67655188-e3bf-4126-b2fa-c5077809aaf4/test_golang args=[]
2020-11-27T17:16:33.000Z [DEBUG] plugin: waiting for RPC address: path=/data/tmp/golang/src/67655188-e3bf-4126-b2fa-c5077809aaf4/test_golang
2020-11-27T17:16:33.266Z [DEBUG] plugin: plugin process exited: path=/data/tmp/golang/src/67655188-e3bf-4126-b2fa-c5077809aaf4/test_golang
2020-11-27T17:16:36.706Z [DEBUG] plugin: starting plugin: path=/data/pipelines/test_golang args=[]
2020-11-27T17:16:36.706Z [DEBUG] plugin: waiting for RPC address: path=/data/pipelines/test_golang
2020-11-27T17:16:36.971Z [DEBUG] plugin: plugin process exited: path=/data/pipelines/test_golang
```

**Comment by Skarlso** [comment]
Ah, I see. I'll try it out later tonight

**Comment by Skarlso** [comment]
Uh, I found the problem. When the docker runner was introduced it wasn't tested with parameterised builds and the docker parameter code is conflicting with the regular parameter things. :/ I'll attempt a nice fix. :)

**Comment by Skarlso** [comment]
@lonnc I found out what was wrong.
I still have to test the fix throughly first, but generally I think I fixed it in this PR: #294
